# Bonus: Temperature Analysis I

In [7]:
import pandas as pd
from datetime import datetime as dt
from dateutil import parser

In [10]:
# "tobs" is "temperature observations"
df = pd.read_csv('Resources/hawaii_measurements.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19550 entries, 0 to 19549
Data columns (total 4 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   station  19550 non-null  object 
 1   date     19550 non-null  object 
 2   prcp     18103 non-null  float64
 3   tobs     19550 non-null  int64  
dtypes: float64(1), int64(1), object(2)
memory usage: 611.1+ KB


In [12]:
# Convert the date column format from string to datetime
df['date']= pd.to_datetime(df['date'])
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19550 entries, 0 to 19549
Data columns (total 4 columns):
 #   Column   Non-Null Count  Dtype         
---  ------   --------------  -----         
 0   station  19550 non-null  object        
 1   date     19550 non-null  datetime64[ns]
 2   prcp     18103 non-null  float64       
 3   tobs     19550 non-null  int64         
dtypes: datetime64[ns](1), float64(1), int64(1), object(1)
memory usage: 611.1+ KB


In [13]:
# Set the date column as the DataFrame index
df.set_index('date')

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73
...                 ...   ...   ...
2017-08-19  USC00516128  0.09    71
2017-08-20  USC00516128   NaN    78
2017-08-21  USC00516128  0.56    76
2017-08-22  USC00516128  0.50    76
2017-08-23  USC00516128  0.45    76

[19550 rows x 3 columns]

In [ ]:
# Drop the date column

### Compare June and December data across all years 

In [50]:
from scipy import stats

In [51]:
# Filter data for desired months
june_df = df[df['date'].dt.month == 6]
december_df = df[df['date'].dt.month == 12]

In [52]:
# Identify the average temperature for June
june_temps_avg = june_df['tobs'].mean()
june_temps_avg

74.94411764705882

In [53]:
# Identify the average temperature for December
december_temps_avg = december_df['tobs'].mean()
december_temps_avg

71.04152933421226

In [56]:
june_temps_df['tobs'].count()

1700

In [57]:
december_temps_df['tobs'].count()

1517

In [58]:
# Create collections of temperature data
june_temps=june_temps_df['tobs']
december_temps=december_temps_df['tobs']

In [59]:
# Run paired t-test
stats.ttest_ind(june_temps, december_temps, equal_var=False)

Ttest_indResult(statistic=31.355036920962423, pvalue=4.193529835915755e-187)

### Analysis

The results of our unpaired t-test show that we can not reject the null hypothesis that the two sets of data are not significantly different from each other. The reason for it is a very high p-value of 4.19. Therefore, we can say that the temperature in Hawaii in December and in June is not significantly different based on the tempreature observations we had from 2010 to 2017. The reason why I chose unpaired t-test is because there was not the same number of datapoints in two datasets we had for different months. We had more measurements for June than for December. 